src/CoreWCF.WebHttp/src/CoreWCF/OpenApi/OpenApiSchemaBuilder.cs

```diff
@@ -193,6 +196,7 @@ private static void PopulateOpenApiPathsOperations(OpenApiDocument document, IEn
 
             string uri = Regex.Replace(operationInfo.UriTemplate, @"\?.*", "");
 
+            uri = address.OriginalString + "/" + uri;
             if (!string.IsNullOrEmpty(additionalBasePath))
```

Comment:
The tests are failing because there are code paths to get here which don't require calling ServiceModelWebServiceBuilderExtensions.AddServiceWebEndpoint, which means OpenApiContractInfo.Address might not be populated. That results in address being null, so a NRE thrown when trying to get address.OriginalString. You need to guard against this possibility by null checking address and only executing this line of code if it's not null. I just did a local validation of this and all the tests were now passing.

Reply:
Thanks, sorry for the late reply, and thanks for the commit.  Not in front of the computer as on a break.